radicle-node/src/service/tracking.rs

```diff
@@ -121,18 +126,23 @@ impl Config {
                         .filter_map(|node| (node.policy == Policy::Track).then_some(node.id))
                         .collect();
 
-                    let ns = if let Ok(repo) = storage.repository(*rid) {
+                    if let Ok(repo) = storage.repository(*rid) {
                         let delegates = repo
                             .delegates()
                             .map_err(|err| FailedDelegates { rid: *rid, err })?
                             .map(PublicKey::from);
                         trusted.extend(delegates);
-                        NonEmpty::from_vec(trusted).map(Namespaces::Many)
                     } else {
-                        Some(Namespaces::All)
+                        // We don't have this repo yet, we should treat the `remote`
+                        // as our seed and include its pub_key(NodeId) in `trusted`.
+                        //
+                        // We are not returning `Namespaces::All` as the `remote`
+                        // might have refs from all kinds of nodes and we might not
+                        // trust all the other nodes.
+                        trusted.push(*remote);
```

Comment:
One problem with this is that we're not guaranteed `remote` is a delegate. So in the case of them not being, we don't have the set of delegates.

This is fine for a stop-gap solution, but I would say add a FIXME mentioning that and we can sort this out in the multistep fetch code.

Reply:
This change is removed.